src/server/databaseoperation/ChefMenuRolloutDatabaseOperator.java

```diff
@@ -0,0 +1,160 @@
+package server.databaseoperation;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+import server.DatabaseConnection;
+import server.model.ChefMenuItemScore;
+import server.model.ChefMenuRollout;
+
+public class ChefMenuRolloutDatabaseOperator {
+
+    private Connection connection;
+    private MenuDatabaseOperator menuDatabaseOperator;
+
+    public ChefMenuRolloutDatabaseOperator() {
+        DatabaseConnection dbInstance = DatabaseConnection.getInstance();
+        this.connection = dbInstance.getConnection();
+        this.menuDatabaseOperator = new MenuDatabaseOperator(); 
+    }
+
+    public boolean insertChefMenuRollout(ChefMenuRollout rollout) throws SQLException {
+        try {
+            int itemId = menuDatabaseOperator.getItemID(rollout.getItemName());
+            int categoryId = menuDatabaseOperator.getCategoryID(rollout.getCategoryName());
+
+            rollout.setItemID(itemId);
+            rollout.setCategoryID(categoryId);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return false;
+        }
+
+        String query = "INSERT INTO ChefMenuRollout (rolloutDate, itemID, categoryID, numberOfVotes) VALUES (?, ?, ?, ?)";
+        try (PreparedStatement stmt = connection.prepareStatement(query)) {
+            stmt.setDate(1, rollout.getRolloutDate());
+            stmt.setInt(2, rollout.getItemID());
+            stmt.setInt(3, rollout.getCategoryID());
+            stmt.setInt(4, rollout.getNumberOfVotes());
+
+            int affectedRows = stmt.executeUpdate();
+            return affectedRows > 0;
+        } catch (SQLException e) {
+            System.err.println("Error inserting ChefMenuRollout: " + e.getMessage());
+            throw e;
```

Comment:
Why are we catching and then throwing back? Are we handling it at the upper level separately or in a specific way?